crates/plugins/src/lib.rs

```diff
@@ -0,0 +1,34 @@
+mod git;
+pub mod install;
+mod plugin_manifest;
+mod prompt;
+pub mod uninstall;
+pub mod version_check;
+
+/// List of Spin internal subcommands
+pub(crate) const SPIN_INTERNAL_COMMANDS: [&str; 9] = [
+    "templates",
+    "up",
+    "new",
+    "bindle",
+    "deploy",
+    "build",
+    "plugin",
+    "trigger",
+    "external",
```

Comment:
if both a `test` internal command and plugin exist, it will be parsed as an internal subcommand and therefore the plugin cannot be invoked until the binary name is changed. Therefore, plugins cannot squat on internal subcommand names.

Reply:
Thanks, that's good to understand.  Unfortunately, I didn't explain my concern clearly (and then muddied the waters with that second thought).  What I was thinking of when I mentioned squatting was something like:

* Alice creates a `spin test` plugin
* Bob likes it and uses it in his CI system
* Now we can't create a built-in `spin test` command because it would break Bob's CI, which would be rude

So if there are commands we're tossing up having, we might want to reserve the names.  Hope that makes more sense now!

Also, might be worth having a way to force mapping to a plugin, e.g. `spin external test` or `spin plugin run test`, so that at least Bob can fix his CI without having to badger Alice to rename her plugin.  That can be retrofitted if the plugin ecosystem merits it though!

Reply:
That makes sense to me. I would assume that if we are releasing a new internal subcommand, the version of spin would change thus not allowing the external plugin to run in the first place without an update to the plugin. This would depend on the version compatibility specified by the plugin.

Although, I like the solution of an `external` subcommand to force run plugins in such cases.

Reply:
My only concern with that would be that we want to avoid plugins having the same name as the internal subcommands. Adding this option would make that test pointless. It might lead to confusion among users due to commands of the same name doing different things.

Reply:
We do!  And it might!  And the test is good because it prevents plugins having the same names as _known_ commands.  The trouble is, we can't stop a plugin having the same name as a command we haven't thought of yet.  So: when we want to use a command name that is already used by an existing plugin, what happens?

As I say... we can probably punt on this for now.  We may need to confront it if the plugin ecosystem takes off, and it would be good to have the shape of a solution ready.  But we don't need to implement it yet.